helpers.go

```diff
@@ -0,0 +1,23 @@
+package par
+
+import (
+	"runtime"
+)
+
+// partsCount returns the number of partitions to use optimised for
+// the available CPUs and the given values.
+func partsCount[In any](values []In) int {
+	if p := runtime.GOMAXPROCS(0); p <= len(values) {
+		return p
+	}
+
+	return len(values)
+}
+
+// parts returns the number of partitions and the size optimised for
+// the available CPUs and given values.
+func parts[In any](values []In) (count, size int) {
```

Comment:
This function needs to know nothing but the length of the slice being passed in, and as such, per Law of Demeter, we should only pass the length, which also removes the need to be generic, and by doing so, the generated code becomes simpler as the function will need 3 less parameters (the size tag of `In` + `cap` and `ptr` of the slice). tThis however goes to the realm of micro-optimizations, but it's a benefit in addition to simplifying the code by restricting boundaries. :)

Reply:
Yep I changed the approach, should have made it take len, but your other suggestion is better ;-)